File: source/protocol/java/test/TestSender.java
import poker_bot.PokerBot.Event;
import poker_bot.PokerBot.Action;

import poker_messaging.PokerMessaging;

class TestSender {
  public static void main(String[] args) throws Exception {
    System.err.println("Start this sender");
    
    Event.Builder message_builder = Event.newBuilder();
    message_builder.setType(Event.Type.JOIN);
    message_builder.setMessage("message 1");
    PokerMessaging.send_message(System.out, message_builder.build());
    
    Event.Builder message_builder2 = Event.newBuilder();
    message_builder2.setType(Event.Type.QUIT);
    message_builder2.setMessage("message 2");
    PokerMessaging.send_message(System.out, message_builder2.build());
    
    PokerMessaging.send_terminator(System.out);
  }
}

File: source/protocol/cpp/test/test_sender.cc
#include "poker_messaging.h"

int main() {
    // Verify that the version of the library that we linked against is
    // compatible with the version of the headers we compiled against.
    GOOGLE_PROTOBUF_VERIFY_VERSION;

    // generate a stream of events
    poker_bot::Event message;
    message.set_type(poker_bot::Event::JOIN);
    message.set_amount(301);
    message.set_message("this is a really long message hopefully longer than 255 bytes");
    if (!send_message<poker_bot::Event>(cout, message)) {
      cerr << "Failed to write message" << endl;
      return -1;
    }
    
    message.set_type(poker_bot::Event::NEW_ROUND);
    message.set_amount(402);
    message.set_message("");
    if (!send_message<poker_bot::Event>(cout, message)) {
      cerr << "Failed to write message" << endl;
      return -1;
    }
    
    send_terminator(cout);

    return 0;
}